Sources/SwiftProtobuf/NameMap.swift

```diff
@@ -91,10 +91,19 @@ public struct _NameMap: ExpressibleByDictionaryLiteral {
   /// has to be computed, it caches the UTF-8 bytes in an
   /// unmovable and immutable heap area.
   internal struct Name: Hashable, CustomStringConvertible {
-    // This is safe to use elsewhere in this library
-    internal init(staticString: StaticString) {
-        self.nameString = .staticString(staticString)
-        self.utf8Buffer = UnsafeRawBufferPointer(start: staticString.utf8Start, count: staticString.utf8CodeUnitCount)
+    // This should not be used outside of this file, as it requires
+    // coordinating the lifecycle with the lifecycle of the pool
+    // where the raw UTF8 gets interned.
+    fileprivate init(staticString: StaticString, pool: InternPool) {
+        if staticString.hasPointerRepresentation {
+            self.nameString = .staticString(staticString)
+            self.utf8Buffer = UnsafeRawBufferPointer(start: staticString.utf8Start, count: staticString.utf8CodeUnitCount)
+        } else {
+            let str = staticString.withUTF8Buffer { String(decoding: $0, as: UTF8.self) }
```

Comment:
This, combined with the interning, is going to end up making two copies: first, a copy will be made to decode the string on this line, and then the `intern` call below will make its own call to `UnsafeMutableRawBufferPointer.allocate` and copy the data from *that* string. The strings themselves will be small (because the pointer rep is only not used for single scalar strings, allegedly), but the extra heap allocations will be slow.

We could instead add a new overload to `intern` that takes a direct UTF8 raw pointer (the one passed to `withUTF8Buffer`) and copies the data out of that, to avoid the extra decoding operation.

Reply:
But don't we still need the first copy because `self.nameString` ends up using `str` directly also?

Reply:
Updated to use the `StaticString` for `nameString` still, and directly copy from the other buffer.